src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```diff
@@ -0,0 +1,141 @@
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ModifyCacheCapacityAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Rca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotClusterSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.BaseClusterRca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.FieldDataCacheClusterRca;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.ShardRequestCacheClusterRca;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+
+// TODO: 1. Read current cache capacity, total cache capacity, upper bound, lower bound from NodeConfigurationRca
+
+public class CacheHealthDecider extends Decider {
+
+  public static final String NAME = "cacheHealthDecider";
+
+  List<String> actionsByUserPriority = new ArrayList<>();
+  private int counter = 0;
+  private ImmutableList<BaseClusterRca> cacheClusterRca;
+
+  public CacheHealthDecider(final long evalIntervalSeconds,
+                            final int decisionFrequency,
+                            final ImmutableList<BaseClusterRca> cacheClusterRca) {
+    // TODO: Also consume NodeConfigurationRca
+    super(evalIntervalSeconds, decisionFrequency);
+    this.cacheClusterRca = cacheClusterRca;
+    configureActionPriority();
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    Decision decision = new Decision(System.currentTimeMillis(), NAME);
+    counter += 1;
+    if (counter < decisionFrequency) {
+      return decision;
+    }
+    counter = 0;
+    cacheClusterRca.forEach(rca -> getActionsFromRca(rca, decision));
+    return decision;
+  }
+
+  private <R extends BaseClusterRca> void getActionsFromRca(final R cacheClusterRca, final Decision decision) {
+    if (!cacheClusterRca.getFlowUnits().isEmpty()) {
+      final HotClusterSummary clusterSummary = cacheClusterRca.getFlowUnits().get(0).getSummary();
+
+      clusterSummary
+              .getHotNodeSummaryList()
+              .forEach(
+                      hotNodeSummary -> {
+                        final NodeKey esNode =
+                                new NodeKey(hotNodeSummary.getNodeID(), hotNodeSummary.getHostAddress());
+                        for (final HotResourceSummary resource :
```

Comment:
do we want to tune multiple resource type simultaneously on the same node instance  ?

Reply:
This is a good point. Perhaps it makes sense to consume heap-health rca also in this decider and ensure we have enough runway to increase cache sizes? Based on the space we have, we can chose to change a single or multiple caches.